project/profiles/capacity4more/modules/c4m/user/c4m_user/includes/c4m_user.theme.inc

```diff
@@ -353,3 +353,54 @@ function theme_c4m_user_contact_link(array $field) {
     ));
   }
 }
+
+/**
+ * Callback for c4m_user_block_contact DS field.
+ *
+ * @param array $field
+ *   The field info.
+ *
+ * @return null|string
+ *   The rendered output.
+ *
+ * @see _contact_personal_tab_access()
+ */
+function theme_c4m_user_block_contact(array $field) {
+  global $user;
+
+  $account = $field['entity'];
+
+  // Anonymous users cannot have contact forms.
+  if (!$account->uid) {
+    return NULL;
+  }
+
+  // Users may not contact themselves.
+  if ($user->uid == $account->uid) {
+    return NULL;
+  }
+
+  // If the requested user has disabled their contact form, or this preference
+  // has not yet been saved, do not allow users to contact them.
+  if (empty($account->data['contact'])) {
+    return NULL;
+  }
+
+  // If requested user has been blocked, do not allow users to contact them.
+  if (empty($account->status)) {
+    return NULL;
+  }
+
+  $group = c4m_og_current_group();
+  if ($group) {
```

Comment:
This field is part of User 'Block List' display.
This display if used also at /people page, which is not within OG context.
If OG context is not checked, og_get_user_roles('node', $group->nid, $user->uid)) will be activated, with $group->nid = NULL, which causes an error.